_posts/2022-08-15-requiring-mfa-on-popular-gems.md

```diff
@@ -0,0 +1,19 @@
+---
+title: Requiring MFA on popular gem maintainers
+layout: post
+author: Jenny Shen
+author_email: jenny.shen@shopify.com
+---
+<p align="center">
+    <img src="/images/gem-with-mfa-flag-dropshadow.png" alt="Doodle of a RubyGem wearing an MFA hat, holding a flag with a checkmark" width="300"/>
+</p>
+
+Two months ago, we outlined our [commitment](https://blog.rubygems.org/2022/06/13/making-packages-more-secure.html) to making Ruby’s supply chain more secure. To combat account takeovers — the second most common software supply chain attack —  we announced a policy to require MFA on at least the top-100 RubyGems packages.
+
+Today (August 15th, 2022), we will begin to enforce MFA on owners of gems with over 180 million total downloads. Users in this category who do not have MFA enabled on the `UI and API` or `UI and gem signin` level will not be able to access profile pages on the web, perform [privileged actions](https://guides.rubygems.org/mfa-requirement-opt-in/#privileged-operations) (i.e. push and yank gems, or add and remove gem owners) or sign in on the command line until they [configure MFA](https://guides.rubygems.org/setting-up-multifactor-authentication/).
```

Comment:
The [PR that implements](https://github.com/rubygems/rubygems.org/pull/3153) this will redirect on all authenticated pages including view access as mentioned in the [RFC](https://github.com/rubygems/rfcs/pull/36/files#diff-3d5cc3acc06fe7e9150fdbfc43399c5ad42572c122187774bfc3a4857df524f1R73).

Since there is a discrepancy, could you elaborate on what you think the behaviour should be on the UI? I was [teetering between all pages vs only edit/create/delete pages](https://github.com/rubygems/rubygems.org/pull/3153#pullrequestreview-1046840259), but we went with all pages because it was simple and effective.